src/main/kotlin/nl/nuts/discovery/service/LocalCertificateAndKeyService.kt

```diff
@@ -69,67 +78,128 @@ class LocalCertificateAndKeyService : CertificateAndKeyService {
     @Autowired
     lateinit var nutsDiscoveryProperties: NutsDiscoveryProperties
 
-    lateinit var certificates: MutableMap<CordaX500Name, X509Certificate>
+    // map of signed certificates
+    lateinit var certificates: MutableMap<CordaX500Name, SignRequest>
+    // map with pending signing requests
+    lateinit var signRequests: MutableMap<CordaX500Name, SignRequest>
 
     @PostConstruct
     fun init() {
         certificates = mutableMapOf()
+        signRequests = mutableMapOf()
     }
 
     override fun submitSigningRequest(request: PKCS10CertificationRequest) {
+        val name = CordaX500Name.parse(request.subject.toString())
```

Comment:
What would be the correct response for a duplicate/conflicting CSR? Does the Corda spec allow something like a 409? And even if so can it be solved by the submitter?

I'm asking since newly signed certs will overwrite old ones with the same name.

Reply:
The way a node can get official feedback is during the polling process.
https://docs.corda.net/setting-up-a-dynamic-compatibility-zone.html#writing-a-doorman-server , not during the post itself. However, the http response is shown during the init step:
So, we can show something for the end-user:
```
Submitting certificate signing request to Corda certificate signing server.
corda_1  | [ERROR] 13:18:23+0000 [main] internal.NodeStartupLogging.invoke - Issue with Node registration: Response Code 404: <html><body><h1>Whitelabel Error Page</h1><p>This application has no explicit mapping for /error, so you are seeing this as a fallback.</p><div id='created'>Mon Dec 02 13:18:23 GMT 2019</div><div>There was an unexpected error (type=Not Found, status=404).</div><div>No message available</div></body></html>
```